### PR TITLE
Add timestamp to log events.

### PR DESCRIPTION
**Pull request by ruflin**
Simplify logging functions

### CODE REVIEW

File: logp/log.go [hunk 1]
```diff
@@ -45,15 +45,20 @@ func debugMessage(calldepth int, selector, format string, v ...interface{}) {
 				return
 			}
 		}
-		if _log.toSyslog {
-			_log.syslog[LOG_INFO].Output(calldepth, fmt.Sprintf(format, v...))
```
Comment (ruflin) [inline logp/log.go]: @tsg Do you know why for syslog LOG_INFO and not debug is used here?

Comment (tsg) [inline logp/log.go]: Hmm, good question. Seems like a bug to me. @monicasarbu any idea?

File: logp/log.go [hunk 2]
```diff
@@ -45,15 +45,20 @@ func debugMessage(calldepth int, selector, format string, v ...interface{}) {
...
-		}
-		if _log.toStderr {
-			_log.logger.Output(calldepth, fmt.Sprintf("DBG  "+format, v...))
-		}
-		if _log.toFile {
-			_log.rotator.WriteLine([]byte(fmt.Sprintf("DBG  "+format, v...)))
-		}
+
+		send(calldepth, LOG_INFO, "DBG  ", format, v)
```
Comment (tsg) [inline logp/log.go]: I think you need `v...` there to pass them as varargs.

File: logp/log.go [hunk 3]
```diff
@@ -73,15 +78,7 @@ func IsDebug(selector string) bool {
...
-		if _log.toSyslog {
-			_log.syslog[level].Output(3, fmt.Sprintf(format, v...))
-		}
-		if _log.toStderr {
-			_log.logger.Output(3, fmt.Sprintf(prefix+format, v...))
-		}
-		if _log.toFile {
-			_log.rotator.WriteLine([]byte(fmt.Sprintf(prefix+format, v...)))
-		}
+		send(3, level, prefix, format, v)
```
Comment (tsg) [inline logp/log.go]: I have a suspicion we need "4" instead of 3 because you added an extra indirection.

File: logp/log.go (hunk 2 above)
Comment (ruflin) [inline logp/log.go]: Fix for both cases where send(...) is used.

File: logp/log.go [hunk 4]
```diff
@@ -45,15 +45,20 @@ func debugMessage(calldepth int, selector, format string, v ...interface{}) {
...
-		}
-		if _log.toFile {
-			_log.rotator.WriteLine([]byte(fmt.Sprintf("DBG  "+format, v...)))
-		}
+
+		send(calldepth, LOG_INFO, "DBG  ", format, v)
+	}
+}
+
+func send(calldepth int, level Priority, prefix string, format string, v ...interface{}) {
```
Comment (urso) [inline logp/log.go]: why not have 'v []interface{}' ?

File: logp/log.go (hunk 2 above)
Comment (urso) [inline logp/log.go]: calldepth + 1? one more function call here

File: logp/log.go (hunk 3 above)
Comment (urso) [inline logp/log.go]: calldepth=4?

Comment (ruflin) [inline logp/log.go]: Good point. Any easy way to test this? I will probably just tryit with one of the beats.

Comment (ruflin) [inline logp/log.go]: BTW: I assume all call levels also for debug have to be incremented.

File: logp/log.go [hunk 5]
```diff
@@ -45,25 +45,30 @@ func debugMessage(calldepth int, selector, format string, v ...interface{}) {
...
-			_log.syslog[LOG_INFO].Output(calldepth, fmt.Sprintf(format, v...))
-		}
-		if _log.toStderr {
-			_log.logger.Output(calldepth, fmt.Sprintf("DBG  "+format, v...))
-		}
-		if _log.toFile {
-			_log.rotator.WriteLine([]byte(fmt.Sprintf("DBG  "+format, v...)))
-		}
+
+		send(calldepth, LOG_INFO, "DBG  ", format, v...)
```
Comment (urso) [inline logp/log.go]: calldepth + 1, send adds another indirection (calldepth is about walking the stack to find matching frame)

**Comment by tsg** [comment]
I gave it a quick try and it seems the file name/number are gone?

<pre>
2015/11/02 12:47:04 DBG  Packet number: [1104]
2015/11/02 12:47:04 DBG  IPv4 packet%!(EXTRA []interface {}=[])
2015/11/02 12:47:04 DBG  TCP packet%!(EXTRA []interface {}=[])
2015/11/02 12:47:04 DBG  Ignore empty non-FIN
</pre>

**Comment by ruflin** [comment]
@tsg Will test it with some beats to see if we get the expected output.

File: logp/log.go [hunk 6]
```diff
@@ -130,24 +127,20 @@ func LogInit(level Priority, prefix string, toSyslog bool, toStderr bool, debugS
...
+		SetToStderr(true, prefix)
 	}
 }
 
 func SetToStderr(toStderr bool, prefix string) {
 	_log.toStderr = toStderr
 	if _log.toStderr {
-		_log.logger = log.New(os.Stderr, prefix, log.Lshortfile)
+		// Add timestamp
+		flag := log.Ldate | log.Ltime | log.LUTC | log.Lshortfile
```
Comment (ruflin) [inline logp/log.go]: @tsg Adding Lshortfile here solves the issue of the non appearing file name.

**Comment by ruflin** [comment]
@tsg Pull request updated and checked with filebeat. Looks good so far.

File: logp/log.go (hunk 1 above)
Comment (monicasarbu) [inline logp/log.go]: @ruflin @tsg it sounds like  bug. It should be LOG_DEBUG.

Comment (ruflin) [inline logp/log.go]: Ok, will push this change into this pull request.

**Comment by monicasarbu** [comment]
LGTM

**Comment by monicasarbu** [comment]
@ruflin @tsg Adding an example with the Filebeat output would be great.

**Comment by tsg** [comment]
Did it with packetbeat just for variety of testing:

<pre>
2015/11/03 17:18:37 sniffer.go:350: DBG  Packet number: 1098
2015/11/03 17:18:37 decoder.go:89: DBG  IPv4 packet
2015/11/03 17:18:37 decoder.go:103: DBG  TCP packet
2015/11/03 17:18:37 sniffer.go:350: DBG  Packet number: 1099
2015/11/03 17:18:37 decoder.go:89: DBG  IPv4 packet
2015/11/03 17:18:37 decoder.go:103: DBG  TCP packet
2015/11/03 17:18:37 sniffer.go:350: DBG  Packet number: 1100
2015/11/03 17:18:37 decoder.go:89: DBG  IPv4 packet
2015/11/03 17:18:37 decoder.go:103: DBG  TCP packet
2015/11/03 17:18:37 sniffer.go:350: DBG  Packet number: 1101
2015/11/03 17:18:37 decoder.go:89: DBG  IPv4 packet
2015/11/03 17:18:37 decoder.go:103: DBG  TCP packet
2015/11/03 17:18:37 sniffer.go:350: DBG  Packet number: 1102
</pre>

Looks pretty good to me, but maybe we want to add millis to it? Happy to also merge it like it is if we want to have it in RC1.

**Comment by andrewkroh** [comment]
Fixes #171

**Pull request merged**